.github/workflows/haskell.yml

```
@@ -198,7 +182,8 @@ jobs:
           find "${{ runner.temp }}/chairman" -type s -exec rm -f {} \;
         fi
 
-    - uses: actions/upload-artifact@v2
+    - name: Save Artifact
+      uses: actions/upload-artifact@v3
```

```suggestion
      uses: actions/upload-artifact@v4
```

There's a v4 version which improves performance: https://github.com/actions/upload-artifact

---

I think this change got lost.